server/src/com/thoughtworks/go/server/service/EnvironmentConfigService.java

```diff
@@ -145,6 +136,24 @@ public EnvironmentConfig named(String environmentName) throws NoSuchEnvironmentE
         return environments.named(new CaseInsensitiveString(environmentName));
     }
 
+    public EnvironmentConfig getEnvironmentConfig(String environmentName) throws NoSuchEnvironmentException {
+        if(environments.hasEnvironmentNamed(new CaseInsensitiveString(environmentName)))
+            return environments.named(new CaseInsensitiveString(environmentName));
+        return null;
```

Comment:
Does not really throw `NoSuchEnvironmentException` is it. There's also a race condition here. Between the time you check `hasEnvironmentNamed` and call `environments.named`, some other thread could have added/removed the environment.

IMO, it's probably best to delegate straight to `environments.named` instead.